java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitutionSolver.java

```diff
@@ -200,16 +205,21 @@ private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol metho
     return substitution;
   }
 
-  private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol method, TypeSubstitution currentSubstitution, boolean isLastParam, JavaType formalType,
-    JavaType argType) {
+  private TypeSubstitution inferTypeSubstitution(JavaSymbol.MethodJavaSymbol method, TypeSubstitution currentSubstitution, JavaType formalType, JavaType argType,
+    boolean isLastParam, List<JavaType> remainingArgTypes) {
     if (formalType.isTagged(JavaType.TYPEVAR)) {
       completeSubstitution(currentSubstitution, formalType, argType);
     } else if (formalType.isArray()) {
       JavaType formalElementType = ((JavaType.ArrayJavaType) formalType).elementType;
       if (argType.isArray()) {
-        completeSubstitution(currentSubstitution, formalElementType, ((JavaType.ArrayJavaType) argType).elementType);
+        TypeSubstitution newSubstitution = inferTypeSubstitution(method, currentSubstitution, formalElementType,
```

Comment:
this if and the else if can be somehow merged with a null check on the arg type computed (element type of array OR lub or remaining args)